From 3b4987bd33af91dd55325da395c236645b0aed16 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 12 Dec 2010 14:48:56 +0000
Subject: [PATCH] mips: Restore dma_cache_sync()

Commit 48e1fd5a81416a037f5a48120bf281102f2584e2 "MIPS: Convert DMA to
use dma-mapping-common.h" renamed and un-exported this function.
Since dma-mapping-common.h does not implement dma_cache_sync(), this
was incorrect.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 arch/mips/mm/dma-default.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/mips/mm/dma-default.c b/arch/mips/mm/dma-default.c
index 4fc1a0f..71d60c4 100644
--- a/arch/mips/mm/dma-default.c
+++ b/arch/mips/mm/dma-default.c
@@ -288,8 +288,8 @@ int mips_dma_supported(struct device *dev, u64 mask)
 	return plat_dma_supported(dev, mask);
 }
 
-void mips_dma_cache_sync(struct device *dev, void *vaddr, size_t size,
-			 enum dma_data_direction direction)
+void dma_cache_sync(struct device *dev, void *vaddr, size_t size,
+		    enum dma_data_direction direction)
 {
 	BUG_ON(direction == DMA_NONE);
 
@@ -298,6 +298,8 @@ void mips_dma_cache_sync(struct device *dev, void *vaddr, size_t size,
 		__dma_sync((unsigned long)vaddr, size, direction);
 }
 
+EXPORT_SYMBOL(dma_cache_sync);
+
 static struct dma_map_ops mips_default_dma_map_ops = {
 	.alloc_coherent = mips_dma_alloc_coherent,
 	.free_coherent = mips_dma_free_coherent,
-- 
1.7.2.3

